### PR TITLE
QE: Move SP migraton to build validation

### PR DESCRIPTION
**Pull request by nodeg**
~~### Waiting for https://github.com/uyuni-project/uyuni/pull/5628 to get merged.~~

## What does this PR change?

This will move the SLE15SP3 to SP4 service pack migration for the minion
to the build validation in order to speed up and simplify the normal
test suite.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage

- Cucumber tests were rearranged

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/18096
Manager 4.3
Manager 4.2

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/run_sets/build_validation_migration.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# This file describes the order of features in a Build Validation test suite run.
```
Comment (vandabarata) [inline testsuite/run_sets/build_validation_migration.yml:1]: If you added it here, shouldn't you need to remove these features from another test run file?

**Review by srbarrios — APPROVED** [review]
It looks good to me.
We can run it as a different stage, once all the smoke tests for all the clients are done.

File: testsuite/run_sets/build_validation_migration.yml (hunk 1 above)
Comment (srbarrios) [reply testsuite/run_sets/build_validation_migration.yml:1]: It's fine, we had them embedded in the init_clients features, on the scenarios removed.

**Review by srbarrios — CHANGES_REQUESTED** [review]
You forget to remove/edit the scenarios from features/reposync/srv_sync_products.feature:

```
  Scenario: Add the initial product for the service pack migration
  Scenario: Installer update channels got enabled when products were added
```

File: testsuite/run_sets/build_validation_migration.yml (hunk 1 above)
Comment (nodeg) [reply testsuite/run_sets/build_validation_migration.yml:1]: I found this description in nearly every .`yml` file for the build validation so I just copied it over to be consistent.

**Comment by nodeg** [comment]
Thanks for the suggestions. I did another commit and addressed them.

**Comment by nodeg** [comment]
I ran the [PR test](https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1516/) for this but then realized that this touches BV and not the regular test suite.

**Review by srbarrios — APPROVED** [review]
It looks good to me, but the only way to be sure it will be to try it :)
Let's merge it, add this new yaml file as new stage in the BV pipeline, and try it on next BV run.

**Review by Bischoff — APPROVED** [review]

**Comment by nodeg** [comment]
Ok, then I will wait until https://github.com/uyuni-project/uyuni/pull/5628 from @Bischoff will get merged, then I will add the `merge-candidate` label here.

**Comment by nodeg** [comment]
Rebase done.

**Pull request merged**